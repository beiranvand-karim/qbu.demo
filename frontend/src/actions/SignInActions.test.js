/* global expect, describe, it, beforeEach */
import configureStore from 'redux-mock-store'
import {
   checkSignInBegin,
   checkSignInSuccess,
   checkSignInError
} from './SignInActions'


describe('SignInActions', () => {

   const mockStore = configureStore();
   const store = mockStore();

   beforeEach(() => {
      store.clearActions();
   });

   it('should matches the check sign in begin snapshot', () => {
      store.dispatch(checkSignInBegin());
      expect(store.getActions()).toMatchSnapshot();
   });

   it('should matches the check sign in success snapshot', () => {
      const token = 'cc0f80d7-ca5c-4c97-aa45-8ee02a5b80de';
      store.dispatch(checkSignInSuccess(token));
      expect(store.getActions()).toMatchSnapshot();
   });

   it('should matches the check sign in error snapshot', () => {
      const error = 'this is a error';
      store.dispatch(checkSignInError(error));
      expect(store.getActions()).toMatchSnapshot();
   });
});
