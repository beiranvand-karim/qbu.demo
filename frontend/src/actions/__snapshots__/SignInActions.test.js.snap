// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInActions should matches the check sign in begin snapshot 1`] = `
Array [
  Object {
    "type": "CHECK_SIGN_IN_BEGIN",
  },
]
`;

exports[`SignInActions should matches the check sign in error snapshot 1`] = `
Array [
  Object {
    "payload": "this is a error",
    "type": "CHECK_SIGN_IN_ERROR",
  },
]
`;

exports[`SignInActions should matches the check sign in success snapshot 1`] = `
Array [
  Object {
    "payload": "cc0f80d7-ca5c-4c97-aa45-8ee02a5b80de",
    "type": "CHECK_SIGN_IN_SUCCESS",
  },
]
`;
